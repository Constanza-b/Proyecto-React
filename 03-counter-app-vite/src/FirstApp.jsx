// export const FirstApp = () => {
//     return <h1>Hola Cami!!</h1>
// }

// const newMessage = {
//   message: "Buenos dÃ­as Cami",
//   title: "Cami te amo mucho!",
// };

import PropTypes from 'prop-types';

export const FirstApp = ({title, subTitle, name}) => {
   
    // console.log(props);
  
    //const suma = (a,b) => a + b;

    return ( 
    <>
      <h1 data-testid="test-title"> { title } </h1>
      {/* <code>{suma(1,6)}</code> */}
      {/* <code>{JSON.stringify(newMessage)}</code> */}
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
    title: PropTypes.string.isRequired, 
    subTitle: PropTypes.string.isRequired 
}

FirstApp.defaultProps = {
    // title: 'No hay titulo',
    subTitle: 'No hay subtitulo',
    name: 'Constanza Becerra'
}