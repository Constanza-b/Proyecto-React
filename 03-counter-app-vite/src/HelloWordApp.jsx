// function App() {
//     return (<h1>Hola mundo!!!</h1>);
// }
// export default App;

//export const HelloWordApp = () => <h1>Hola Cony</h1>;

export const HelloWordApp = () => {
  return (
    <h1>Hello Word App</h1>
  )
}

