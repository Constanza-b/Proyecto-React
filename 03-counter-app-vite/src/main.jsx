import React from 'react';
import ReactDOM from 'react-dom/client';
// import { HelloWordApp } from './HelloWordApp';
// import { FirstApp } from './FirstApp';
import { CounterApp } from './CounterApp';
import './styles.css';




ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* <HelloWordApp/> */}
        {/* <FirstApp title='Hola, soy Cony' /> */}
        <CounterApp value = {10}/>
    </React.StrictMode>
);