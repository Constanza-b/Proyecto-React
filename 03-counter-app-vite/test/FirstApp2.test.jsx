import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Pruebas en <FirstApp /> ", () => {
  const title = "Hola, soy Cony";
  const subTitle = "Soy un subtitulo";

  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('Debe mostrar el mensaje "Hola, soy Cony"', () => {
    render(<FirstApp title={title} />);
    // screen.debug();
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("Debe mostrar el tÃ­tulo en un h1", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test("Debe mostrar el subtitulo enviado por props", () => {
    render(
      <FirstApp 
        title={title} 
        subTitle={subTitle} 
      />
    );
    expect(screen.getAllByText(subTitle).length).toBe(2);
  });
});
