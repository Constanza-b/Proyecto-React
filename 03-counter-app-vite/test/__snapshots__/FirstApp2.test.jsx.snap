// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp />  Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, soy Cony
     
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Constanza Becerra
  </p>
</div>
`;
